src/components/Header/index.tsx

```diff
@@ -1,54 +1,103 @@
-}
+  mode?: HeaderMode;
+  children?: React.ReactNode;
+  className?: string;
+};
+
+const navigation = [
+  {
+    href: "/news",
+    name: "News"
+  },
+  {
+    href: "/features",
+    name: "Features"
+  },
+  {
+    href: "https://github.com/YARC-Official/YARC-Launcher/releases/tag/v0.3.1",
+    name: "Download",
+    redirect: true
+  },
+  {
+    href: "https://yarg.youtrack.cloud/issues",
+    name: "Roadmap",
+    redirect: true
+  },
+  {
+    href: "/community",
+    name: "Community"
+  }
+];
 
 const Header: React.FC<Props> = ({
-    mode = "dark",
-    children,
-    className,
+  mode = "dark",
+  children,
+  className,
 }: Props) => {
 
-    return <div className={[styles.header, className].join(" ")}>
-        <header data-mode={mode} className={styles.bar}>
-            <Link href={"/"}>
-                <YARGLogo className={styles.logo} />
-            </Link>
-            <Navigation />
-        </header>
+  const [searchValue, setSearchValue] = useState<string>("");
+
+  // Temporary mocks
+  const isLogged = false;
+  const mockedBannerMessage = "Lorem ipsum dolor sit amet.";
+  const mockedTwitchBannerHref = "https://www.twitch.tv/eliteasian123";
+  const mockedTwitchBannerMessage = "EliteAsian123 is LIVE on Twitch! Working on Engine rewrite";
+
+  const handleSearchConfirm = () => {
+    // do something on confirm
+    console.log(searchValue);
+  };
 
-        { children }
+  const handleBannerClick = () => {};
+
+  return (
+    <div className={[styles.header, className].join(" ")}>
+      {/* <HeaderBanner.Warning message={mockedBannerMessage} /> */}
+      {/* <HeaderBanner.Announcement message={mockedBannerMessage} onClick={handleBannerClick} /> */}
+      {/* <HeaderBanner.Live href={mockedTwitchBannerHref} message={mockedTwitchBannerMessage} /> */}
+      
+      <header data-mode={mode} className={styles.bar}>
+        <HeaderSearch value={searchValue} handleChange={setSearchValue} handleConfirm={handleSearchConfirm} />
+        <Navigation />
+        {isLogged ? <LoggedInComponents /> : null}
+      </header>
+      {children}
     </div>
-}
+  );
+};
 
 function Navigation() {
-    return (<>
-        <ul>
-            {/* 
-                TO-DO: These pages still not developed.
-                <li>Features</li>
-                <li>Downloads</li>
-                <li>Community</li>
-                <li>News</li>
-            */}
-
-            <a href="https://yarg.youtrack.cloud/issues" target="_blank">
-                <li>Roadmap</li>
-            </a>
-
-            <a href="https://docs.yarg.in/" target="_blank">
-                <li>Documentation</li>
-            </a>
-        </ul> 
-    </>);
-}
-
-export default Header;
\ No newline at end of file
+  return (
+    <ul>  
+      <NavItem label="Home" href="/" />
+      {navigation.map(({ href, name, redirect }) => (
+        <NavItem key={crypto.randomUUID()} href={href} label={name} target={redirect ? "_blank" : "_self"} /> 
```

Comment:
The key should be match between renders, so generating random number with crypto on the fly it's not good practice, because react will lose the reference of the list item and bugs may appear.

My suggestion is define a unique id to each element of `navigation`, as it is not yet dynamic, so you have full control over whether you define an id or a unique attribute.

If you want to read more about https://react.dev/learn/rendering-lists#:~:text=Similarly%2C%20do%20not,on%20the%20data.